x/distribution/keeper/grpc_query.go

```diff
@@ -361,7 +362,11 @@ func (k Querier) DelegatorWithdrawAddress(ctx context.Context, req *types.QueryD
 }
 
 // CommunityPool queries the community pool coins
-func (k Querier) CommunityPool(ctx context.Context, req *types.QueryCommunityPoolRequest) (*types.QueryCommunityPoolResponse, error) {
+func (k Querier) CommunityPool(ctx context.Context, _ *types.QueryCommunityPoolRequest) (*types.QueryCommunityPoolResponse, error) {
+	if k.HasExternalCommunityPool() {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "external community pool is enabled - use the CommunityPool query exposed by the external community pool")
+	}
+
```

Comment:
have we factored out the possibility of querying the pool and returning it here? or can we not go from Coins to DecCoins?

Reply:
do you mean querying the `ExternalCommunityPool` from `x/distribution`?

Reply:
yeah, instead of erroring, have a query method on the external pool interface and invoke it here

Reply:
@re discussion: we should make a comment on the RPC in the proto file of x/distribution that this function will error if external community pool is set

we should also do this for the txs that fail if its set